src/modules/auth/csrf.utils.ts

```diff
@@ -11,7 +13,11 @@ export const checkClientCsrf = (req: Request, res: Response, next: NextFunction)
             }
         })
         csrf(req, res, next)
-    } else {
+    } else if (req.headers['server-request'] === process.env.CSRF_SERVER_SECRET) {
```

Comment:
this way server requests will be validated by sending process.env.CSRF_SERVER_SECRET with headers when it's a server call, so hackers not only have to send server-request header but also have to guess the process.env.CSRF_SERVER_SECRET from .env file.